main.py

```diff
@@ -35,10 +43,20 @@
                 case pygame.K_UP:
                     if y - movementWidth / 2 > 0:
                         y -= movementWidth
+        # collect coin
+        elif player.colliderect(coin):
+            coinCollected = False
 
-    # change background color and draw player
+    # change background color
     window.fill((0, 0, 0))
-    pygame.draw.rect(window, (0, 0, 255), (x, y, 40, 40))
+
+    # draw coin
+    if coinCollected:
+        pygame.draw.circle(window, (212, 175, 55), (100, 100), 16)
+
+    # draw player
+    player = pygame.Rect(x, y, 40, 40)
```

Comment:
I don't see the advantage of initializing the player object before the loop, when it is overridden here in the first iteration of the loop anyway. Effectively, the player coordinates are now stored twice seperately (in the `player` object and in the variables `x` and `y`), so I'm not sure if the `player` object brings any advantage.